src/complex/Utilities/StringUtilities.hpp

```diff
@@ -193,5 +199,20 @@ inline std::string simplified(std::string_view text)
   }
   return finalString;
 }
+
+inline std::string toUpper(const std::string& input)
+{
+  std::string data = input;
+  std::for_each(data.begin(), data.end(), [](char& c) { c = ::toupper(c); });
+  return data;
+}
+
+inline std::string toLower(const std::string& input)
+{
+  std::string data = input;
+  std::for_each(data.begin(), data.end(), [](char& c) { c = ::tolower(c); });
+  return data;
+}
```

Comment:
```cpp
/**
 * @brief Converts a char to its uppercase version according to the current locale.
 * Uses std::toupper internally.
 * @param c
 * @return char
 */
inline char toUpper(char c)
{
  return static_cast<char>(std::toupper(static_cast<unsigned char>(c)));
}

/**
 * @brief Converts a char to its lowercase version according to the current locale.
 * Uses std::tolower internally.
 * @param c
 * @return char
 */
inline char toLower(char c)
{
  return static_cast<char>(std::tolower(static_cast<unsigned char>(c)));
}

/**
 * @brief Converts the given string to uppercase.
 * Assumes that the string is ASCII.
 * @param input
 * @return std::string
 */
inline std::string toUpper(std::string input)
{
  for(char& c : input)
  {
    c = toUpper(c);
  }
  return input;
}

/**
 * @brief Converts the given string to lowercase.
 * Assumes that the string is ASCII.
 * @param input
 * @return std::string
 */
inline std::string toUpper(std::string input)
{
  for(char& c : input)
  {
    c = toLower(c);
  }
  return input;
}
```
I think making our own wrapper for `std::toupper` and `std::tolower` makes things cleaner with less casting. And with range based for loops, we don't have to include `<algorithm>` for `std::for_each`.

Reply:
Changes have been addressed in latest commits